docs/quick-starts/rug-typescript.md

```diff
@@ -0,0 +1,202 @@
+$ rug test
+Resolving dependencies for russmiles:scattered-rugs:0.1.0 ← local completed
+Loading russmiles:scattered-rugs:0.1.0 ← local into runtime completed
+Running test scenarios in russmiles:scattered-rugs:0.1.0 ← local completed
+
+→ Failed Scenarios
+  SimpleSampleEditor should just add a file called "README.md" to the target project (1 of 1 assertions failed)
+    Failed Assertions
+       Scenario 'SimpleSampleEditor should just add a file called "README.md" to the target project' tests editor 'SimpleSampleEditor', which was not found.
+Known operations are []
+```
+
+With our test *happily* failing you can now write the following Rug TypeScript editor to meet the assertions of the test. Create a file called `SimpleSampleEditor.ts` in the `.atomist/editors` directory that contains the following:
+
+```
+import { ProjectEditor } from "@atomist/rug/operations/ProjectEditor"
+import { Status, Result, Parameter } from "@atomist/rug/operations/RugOperation"
+import { Project, Pair, File } from '@atomist/rug/model/Core'
+import { PathExpression, PathExpressionEngine, TreeNode, Match } from '@atomist/rug/tree/PathExpression'
+
+
+let params: Parameter[] = [
+    {
+        name: "description",
+        displayName: "Description",
+        description: "Text to be added to the README.md",
+        validInput: "Any valid text",
+        minLength: 1,
+        pattern: "@any",
+    }
+]
+
+interface Parameters {
+    description: string
+}
+
+export let editor: ProjectEditor = {
+    tags: ["simple"],
+    name: "SimpleSampleEditor",
+    description: "A simple sample TypeScript editor",
+    parameters: params,
+    edit(project: Project, p: Parameters): Result {
+
+        project.addFile("README.md", p.description);
+
+        return new Result(Status.Success, "README.md added to project")
+    }
+}
+```
+
+Walking through this editor the contents are:
+
+-   Importing the TypeScript types for working with Rug.
+-   Describing a single, mandatory parameter `description` to this editor.
+-   Tagging and exporting a `ProjectEditor` implementation that contains the `edit` function implementation for the project-editing logic.
+
+You should now be able to execute `rug test` from your project's root directory and get a similar output to the following:
+
+```
+$ rug test
+Resolving dependencies for russmiles:scattered-rugs:0.1.0 ← local completed
+Loading russmiles:scattered-rugs:0.1.0 ← local into runtime completed
+Executing scenario SimpleSampleEditor should just add a file called "README.md" to the target project...      
+  Testing assertion fileExists(SimpleLiteral(README.md))                                                      
+  Testing assertion fileContains(SimpleLiteral(README.md),SimpleLiteral(Hello, Rug TypeScript World!))        
+Running test scenarios in russmiles:scattered-rugs:0.1.0 ← local completed
+
+Successfully executed 1 of 1 scenarios: Test SUCCESS
+```
+
+### More Information
+
+That's it for our brief introduction to Rug
```

Comment:
Do you think "Rug TypeScript" is a thing, or is it just using TypeScript to create Rugs?

I just wonder if Rug JavaScript, Rug ClojureScript etc. would sound ok...